enterprise/server/cmd/ci_runner/main.go

```diff
@@ -122,6 +123,127 @@ type workspace struct {
 	logBuf bytes.Buffer
 }
 
+type buildEventReporter struct {
```

Comment:
Could we move this to a separate package or does it need to be in here?

Reply:
Nothing is forcing us to leave this here, but also not a lot of motivation for moving it out. For now it's somewhat convenient to have it next to the runner code.